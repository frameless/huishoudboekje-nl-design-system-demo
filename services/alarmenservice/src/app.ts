import cors from "cors";
import express from "express";
import apiV1Router from "./api-v1";
import {errorHandler} from "./errorHandlers";
import healthRouter from "./health";

const app = express();
const DATABASE_URL = process.env.DATABASE_URL;

if (!DATABASE_URL && process.env.NODE_ENV !== "test") {
	throw new Error("Unknown database. Please provide the database connection string in DATABASE_URL in the environment.");
}

const logger = () => (req, res, next) => {
	console.log("[EVENT]", `Incoming request: ${req.method} ${req.url}`);
	if (req.body) {
		console.log("[EVENT]", "Body:", req.body);
	}
	next();
};

app.use(cors());
app.use(express.json());
if (process.env.NODE_ENV === 'dev') {
	app.use(logger());
}

app.get("/health", healthRouter);
app.use("/v1/alarms", apiV1Router);
app.use(errorHandler);

export default app;