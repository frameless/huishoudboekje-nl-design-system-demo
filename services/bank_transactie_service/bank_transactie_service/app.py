""" Main app for bank_transactie_service """
import logging
import logging.config

from flask import Flask, Response
from flask_migrate import Migrate

from bank_transactie_service.views.bank_transaction import BankTransactionView
from bank_transactie_service.views.customer_statement_message import CustomerStatementMessageView
from core_service import database
from bank_transactie_service.views.bank_transaction_range import BanktransactionRangeView
from bank_transactie_service.views.transactions_filter_view import BanktransactionFilterView
from bank_transactie_service.views.bank_transaction_sum import BanktransactionSumView
from core_service.sqlalchemy_statsd_metrics import add_sqlalchemy_statsd_metrics

db = database.db


def create_app(config_name='bank_transactie_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    db.init_app(app)
    Migrate(app, db)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    # logging.config.dictConfig(
    #     {
    #         "version": 1,
    #         "incremental": True,
    #         "loggers": {"sqlalchemy.engine": {"level": app.config["LOG_LEVEL"]}},
    #     }
    # )
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel(app.config["LOG_LEVEL"])

    add_sqlalchemy_statsd_metrics(app)


    @app.route('/health')
    def health():
        return Response()

    # Views
    routes = [
        {"path": "/customerstatementmessages", "view": CustomerStatementMessageView,
            "name": "customer_statement_message_view"},
        {"path": "/customerstatementmessages/<object_id>", "view": CustomerStatementMessageView,
            "name": "customer_statement_message_detail_view"},
        {"path": "/banktransactions", "view": BankTransactionView,
            "name": "banktransaction_view"},
        {"path": "/banktransactions/<object_id>", "view": BankTransactionView,
            "name": "banktransaction_detail_view"},
        {"path": "/banktransactions/filter", "view": BanktransactionFilterView,
            "name": "banktransaction_filter_view"},
        {"path": "/banktransactions/range", "view": BanktransactionRangeView,
            "name": "banktransaction_range_view"},
        {"path": "/banktransactions/sum", "view": BanktransactionSumView,
            "name": "banktransaction_sum_view"},
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app


if __name__ == '__main__':
    create_app().run()
