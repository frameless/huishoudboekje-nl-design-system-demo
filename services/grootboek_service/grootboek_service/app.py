""" Main app for grootboek_service """
from models import *
import logging
from flask import Flask, Response
from flask_migrate import Migrate
from core_service import database
from grootboek_service.views.grootboekrekeningen import GrootboekrekeningenView
from core_service.statsd_metrics import add_statsd_metrics

db = database.db


def create_app(config_name='grootboek_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel(app.config["LOG_LEVEL"])

    db.init_app(app)
    Migrate(app, db)

    add_statsd_metrics(app)

    @app.route('/health')
    def health():
        return Response()

    # Views
    routes = [
        {"path": "/grootboekrekeningen", "view": GrootboekrekeningenView,
         "name": "grootboekrekening_view"},
        {"path": "/grootboekrekeningen/<object_id>", "view": GrootboekrekeningenView,
         "name": "grootboekrekening_detail_view"}
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app


if __name__ == '__main__':
    create_app().run()
