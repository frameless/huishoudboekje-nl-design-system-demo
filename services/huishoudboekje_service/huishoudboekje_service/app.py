""" Main app for huishoudboekje_service """
import logging
import os
from flask import Flask, Response
from flask_migrate import Migrate

from huishoudboekje_service.views import (
    BurgerView,
    AfspraakView,
    RekeningView,
    RekeningBurgerView,
    RekeningAfdelingView,
    JournaalpostView,
    RubriekView,
    ConfiguratieView,
    OverschrijvingView,
    ExportView,
    HuishoudenView,
    AfdelingView,
)
from core_service import database

db = database.db


def create_app(config_name='huishoudboekje_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    logging.info(f"Starting {__name__} with {config_name}")

    db.init_app(app)
    migrate = Migrate(app, db)

    @app.route('/health')
    def health():
        return Response()

    # Views
    routes = [
        {"path": "/burgers", "view": BurgerView, "name": "burger_view"},
        {"path": "/burgers/<object_id>", "view": BurgerView, "name": "burger_detail_view"},
        {"path": "/burgers/<object_id>/rekeningen", "view": RekeningBurgerView,
         "name": "burger_rekeningen_view"},
        {"path": "/afspraken", "view": AfspraakView, "name": "afspraak_view"},
        {"path": "/afspraken/<object_id>", "view": AfspraakView, "name": "afspraak_detail_view"},
        {"path": "/rekeningen", "view": RekeningView, "name": "rekening_view"},
        {"path": "/rekeningen/<object_id>", "view": RekeningView, "name": "rekening_detail_view"},
        {"path": "/journaalposten", "view": JournaalpostView, "name": "journaalpost_view"},
        {"path": "/journaalposten/<object_id>", "view": JournaalpostView, "name": "journaalpost_detail_view"},
        {"path": "/rubrieken", "view": RubriekView, "name": "rubriek_view"},
        {"path": "/rubrieken/<object_id>", "view": RubriekView, "name": "rubriek_detail_view"},
        {"path": "/configuratie", "view": ConfiguratieView, "name": "configuratie_view"},
        {"path": "/configuratie/<object_id>", "view": ConfiguratieView, "name": "configuratie_detail_view"},
        {"path": "/overschrijvingen", "view": OverschrijvingView, "name": "overschrijving_view"},
        {"path": "/overschrijvingen/<object_id>", "view": OverschrijvingView, "name": "overschrijving_detail_view"},
        {"path": "/export", "view": ExportView, "name": "export_view"},
        {"path": "/export/<object_id>", "view": ExportView, "name": "export_detail_view"},
        {"path": "/huishoudens", "view": HuishoudenView, "name": "huishouden_view"},
        {"path": "/huishoudens/<object_id>", "view": HuishoudenView, "name": "huishouden_detail_view"},
        {"path": "/afdelingen", "view": AfdelingView, "name": "afdeling_view"},
        {"path": "/afdelingen/<object_id>", "view": AfdelingView, "name": "afdeling_detail_view"},
        {"path": "/afdelingen/<object_id>/rekeningen", "view": RekeningAfdelingView,
         "name": "afdeling_rekeningen_view"},
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app
