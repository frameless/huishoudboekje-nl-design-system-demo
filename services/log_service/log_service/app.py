""" Main app for log_service """
import logging
import logging.config

from flask import Flask, Response
from flask_migrate import Migrate

from core_service import database
from log_service.views.gebruikersactiviteit import GebruikersActiviteitView
from core_service.statsd_metrics import add_statsd_metrics

db = database.db


def create_app(
    config_name="log_service.config.Config"
):
    app = Flask(__name__)
    app.config.from_object(config_name)

    db.init_app(app)
    Migrate(app, db)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    # logging.config.dictConfig(
    #     {
    #         "version": 1,
    #         "incremental": True,
    #         "loggers": {"sqlalchemy.engine": {"level": app.config["LOG_LEVEL"]}},
    #     }
    # )
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel("WARNING")

    add_statsd_metrics(app)

    @app.route("/health")
    def health():
        return Response()

    # Views
    routes = [
        {
            "path": "/gebruikersactiviteiten",
            "view": GebruikersActiviteitView,
            "name": "gebruikersactiviteit_view",
        },
        {
            "path": "/gebruikersactiviteiten/<object_id>",
            "view": GebruikersActiviteitView,
            "name": "gebruikersactiviteit_detail_view",
        },
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False,
        )

    return app


if __name__ == "__main__":
    create_app().run()
