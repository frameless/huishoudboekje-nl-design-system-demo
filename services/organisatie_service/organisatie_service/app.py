""" Main app for organisatie_service """
from models import *
import logging

from flask import Flask, Response
from flask_migrate import Migrate

from organisatie_service.views.organisaties import OrganisatieView
from organisatie_service.views.organisatie_rekeningen import OrganisatieRekeningenView
from organisatie_service.views.afdelingen import AfdelingView
from core_service import database
from core_service.sqlalchemy_statsd_metrics import add_sqlalchemy_statsd_metrics

db = database.db


def create_app(config_name='organisatie_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel(app.config["LOG_LEVEL"])

    db.init_app(app)
    Migrate(app, db)

    add_sqlalchemy_statsd_metrics(app)

    @app.route('/health')
    def health():
        return Response()

    routes = [
        {"path": "/organisaties", "view": OrganisatieView, "name": "organisatie_view"},
        {"path": "/organisaties/<object_id>", "view": OrganisatieView,
            "name": "organisatie_detail_view"},
        {"path": "/organisaties/rekeningen", "view": OrganisatieRekeningenView, "name": "organisatie_rekeningen_view"},
        {"path": "/afdelingen", "view": AfdelingView, "name": "afdeling_view"},
        {"path": "/afdelingen/<object_id>", "view": AfdelingView,
            "name": "afdeling_detail_view"},
    ]

    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app


if __name__ == '__main__':
    create_app().run()
