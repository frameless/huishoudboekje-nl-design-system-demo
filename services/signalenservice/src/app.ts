import cors from "cors";
import express from "express";
import apiV1Router from "./api-v1";
import {errorHandler} from "./errorHandlers";
import healthRouter from "./health";
import log from "loglevel";

const app = express();
const DATABASE_URL = process.env.DATABASE_URL;

if (!DATABASE_URL && process.env.NODE_ENV !== "test") {
	throw new Error("Unknown database. Please provide the database connection string in DATABASE_URL in the environment.");
}

const logger = () => (req, res, next) => {
	log.debug("[EVENT]", `Incoming request: ${req.method} ${req.url}`);
	if (req.body) {
		log.debug("[EVENT]", "Body:", req.body);
	}
	next();
};

app.use(cors());
app.use(express.json());
if (process.env.NODE_ENV === 'dev') {
	app.use(logger());
}

app.get("/health", healthRouter);
app.use("/v1/signals", apiV1Router);
app.use(errorHandler);

export default app;