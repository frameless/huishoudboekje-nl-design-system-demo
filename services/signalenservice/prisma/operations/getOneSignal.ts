import prisma from "../../src/client";
import {NotFoundError} from "../../src/errorHandlers";

const getOneSignal = async (id: string) => {
	const data = await prisma.signal.findUnique({
		where: {id},
	});

	if (!data) {
		throw new NotFoundError();
	}

	return data;
};

export default getOneSignal;