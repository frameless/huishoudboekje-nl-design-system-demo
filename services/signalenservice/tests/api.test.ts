import {Prisma, Signal} from "@prisma/client";
import supertest from "supertest";
import {prismaMock} from "../.jest/mockClient";
import getOneSignal from "../prisma/operations/getOneSignal";
import app from "../src/app";
import signals from "./fixtures/signals.json";
import ResolvedValue = jest.ResolvedValue;

const api = supertest(app);

const toSignal = (data: any): Signal => {
	return {
		...data,
	};
};

describe("Signal CRUD (operations)", () => {

	describe("Given the Signal exists", () => {
		it("should return the Signal", async () => {
			const signal = toSignal(signals[0]);
			prismaMock.signal.findUnique.mockResolvedValue(signal);

			const result = await api.get("/v1/signals/" + signal.id);
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should return all Signals", async () => {
			prismaMock.signal.findMany.mockResolvedValue(signals as ResolvedValue<unknown>);

			const result = await api.get("/v1/signals/");
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should return Signals filtered by id (through request query params)", async () => {
			const filterIds = signals.map(a => a.id).slice(0, 2);
			prismaMock.signal.findMany.mockResolvedValue(signals.filter(a => filterIds.includes(a.id)) as ResolvedValue<unknown>);

			const result = await api.get("/v1/signals/?filter_ids=" + filterIds.join(","));
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should return Signals filtered by id (through request body)", async () => {
			const filterIds = signals.map(a => a.id).slice(0, 2);
			prismaMock.signal.findMany.mockResolvedValue(signals.filter(a => filterIds.includes(a.id)) as ResolvedValue<unknown>);

			const result = await api.get("/v1/signals/").send({
				filter_ids: filterIds,
			});
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should update the Signal", async () => {
			const newSignal: Omit<Signal, "id"> = {
				alarmId: "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
				banktransactieIds: [],
				isActive: true,
				type: "default",
				actions: [],
				context: {
					is: true,
					something: "else",
				},
				bedragDifference: "300",
				timeUpdated: new Date("2022-09-01T23:00:00"),
			};
			const timeUpdated = signals[0].timeUpdated;

			prismaMock.signal.update.mockResolvedValue({
				...newSignal,
				id: signals[0].id,
				timeUpdated,
			} as ResolvedValue<unknown>);

			const result = await api.put("/v1/signals/" + signals[0].id).send(newSignal);

			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should delete the Signal", async () => {
			prismaMock.signal.delete.mockResolvedValue(undefined!);

			const result = await api.delete("/v1/signals/" + signals[0].id);
			expect(result.statusCode).toBe(204);
		});
	});

	describe("Given the Signal does not exist", () => {
		it("should return 404 Not Found when tyring to getOneSignal", async () => {
			prismaMock.signal.update.mockRejectedValue(new Prisma.PrismaClientKnownRequestError("Not found", {code: "P2025", clientVersion: "0.0.0"}));

			const result = await api.get("/v1/signals/" + signals[0].id);
			expect(result.statusCode).toEqual(404);
		});

		it("should create the Signal", async () => {
			const {id, ...signal} = toSignal(signals[0]);
			prismaMock.signal.create.mockResolvedValue({...signal, id});

			const result = await api.post("/v1/signals").send(signal);
			expect(result.statusCode).toBe(201);
			expect(result.body).toMatchSnapshot();
		});

		it("should return 404 Not Found when trying to updateSignal", async () => {
			const newStreet = "Barstreet";
			prismaMock.signal.update.mockRejectedValue(new Prisma.PrismaClientKnownRequestError("Not found", {code: "P2025", clientVersion: "0.0.0"}));

			const result = await api.put("/v1/signals/" + signals[0].id).send({
				id: signals[0].id,
				street: newStreet,
			});
			expect(result.statusCode).toEqual(404);
		});

		it("should delete the Signal", async () => {
			prismaMock.signal.delete.mockRejectedValue(new Prisma.PrismaClientKnownRequestError("Not found", {code: "P2025", clientVersion: "0.0.0"}));
			const result = await api.delete("/v1/signals/" + signals[0].id);
			expect(result.statusCode).toEqual(204);
		});
	});

	describe("Given an unknown error occurs on the server", () => {
		const error = new Error("Server did a boo boo");

		it("should return a 500 when tyring to getOneSignal", async () => {
			prismaMock.signal.findUnique.mockRejectedValue(error);
			const getSignalResult = getOneSignal(signals[0].id);
			await expect(getSignalResult).rejects.toEqual(error);

			const result = await api.get("/v1/signals/" + signals[0].id);
			expect(result.statusCode).toEqual(500);
		});

		it("should return a 500 when tyring to getManysignals", async () => {
			prismaMock.signal.findMany.mockRejectedValue(error);
			const ids = signals.map(a => a.id);

			const result = await api.get("/v1/signals/?filter_ids=" + ids.join(","));
			expect(result.statusCode).toEqual(500);
		});

		it("should return a 500 when trying to create the Signal", async () => {
			prismaMock.signal.create.mockRejectedValue(error);

			const result = await api.post("/v1/signals").send(signals[0]);
			expect(result.statusCode).toBe(500);
			expect(result.body).toMatchSnapshot();
		});

		it("should return a 500 when trying to updateSignal", async () => {
			prismaMock.signal.update.mockRejectedValue(error);

			const newStreet = "Barstreet";
			const result = await api.put("/v1/signals/" + signals[0].id).send({
				id: signals[0].id,
				street: newStreet,
			});
			expect(result.statusCode).toEqual(500);
		});

		it("should return a 500 when trying to deleteSignal", async () => {
			prismaMock.signal.delete.mockRejectedValue(error);
			const result = await api.delete("/v1/signals/" + signals[0].id);
			expect(result.statusCode).toEqual(500);
		});
	});

	describe("Given the server is healthy", () => {
		it("should return 200 OK", async () => {
			const result = await api.get("/health");
			expect(result.statusCode).toEqual(200);
		});
	});

});
