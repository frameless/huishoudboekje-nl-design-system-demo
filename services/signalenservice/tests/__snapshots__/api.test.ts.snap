// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signal CRUD (operations) Given an unknown error occurs on the server should return a 500 when trying to create the Signal 1`] = `
{
  "message": "Server did a boo boo",
  "ok": false,
}
`;

exports[`Signal CRUD (operations) Given the Signal does not exist should create the Signal 1`] = `
{
  "data": {
    "actions": [],
    "alarmId": "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
    "banktransactieIds": [],
    "bedragDifference": "100",
    "context": {
      "is": true,
      "something": "else",
    },
    "id": "02c49b0b-def0-46b2-ae03-9b90430adb94",
    "isActive": true,
    "timeUpdated": "2022-05-16T13:00:00.000Z",
    "type": "default",
  },
  "ok": true,
}
`;

exports[`Signal CRUD (operations) Given the Signal exists should return Signals filtered by id (through request body) 1`] = `
{
  "data": [
    {
      "actions": [],
      "alarmId": "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
      "banktransactieIds": [],
      "bedragDifference": "100",
      "context": {
        "is": true,
        "something": "else",
      },
      "id": "02c49b0b-def0-46b2-ae03-9b90430adb94",
      "isActive": true,
      "timeUpdated": "2022-05-16T13:00:00.000Z",
      "type": "default",
    },
    {
      "actions": [],
      "alarmId": "5d65d8e9-1f4a-4cb9-a89c-0f952ac5f2ff",
      "banktransactieIds": [],
      "bedragDifference": null,
      "context": {
        "is": false,
        "something": "else",
      },
      "id": "bef7c096-40b6-4158-b0a5-5b6b813080be",
      "isActive": true,
      "timeUpdated": "2022-05-16T13:00:00.000Z",
      "type": "default",
    },
  ],
  "ok": true,
}
`;

exports[`Signal CRUD (operations) Given the Signal exists should return Signals filtered by id (through request query params) 1`] = `
{
  "data": [
    {
      "actions": [],
      "alarmId": "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
      "banktransactieIds": [],
      "bedragDifference": "100",
      "context": {
        "is": true,
        "something": "else",
      },
      "id": "02c49b0b-def0-46b2-ae03-9b90430adb94",
      "isActive": true,
      "timeUpdated": "2022-05-16T13:00:00.000Z",
      "type": "default",
    },
    {
      "actions": [],
      "alarmId": "5d65d8e9-1f4a-4cb9-a89c-0f952ac5f2ff",
      "banktransactieIds": [],
      "bedragDifference": null,
      "context": {
        "is": false,
        "something": "else",
      },
      "id": "bef7c096-40b6-4158-b0a5-5b6b813080be",
      "isActive": true,
      "timeUpdated": "2022-05-16T13:00:00.000Z",
      "type": "default",
    },
  ],
  "ok": true,
}
`;

exports[`Signal CRUD (operations) Given the Signal exists should return all Signals 1`] = `
{
  "data": [
    {
      "actions": [],
      "alarmId": "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
      "banktransactieIds": [],
      "bedragDifference": "100",
      "context": {
        "is": true,
        "something": "else",
      },
      "id": "02c49b0b-def0-46b2-ae03-9b90430adb94",
      "isActive": true,
      "timeUpdated": "2022-05-16T13:00:00.000Z",
      "type": "default",
    },
    {
      "actions": [],
      "alarmId": "5d65d8e9-1f4a-4cb9-a89c-0f952ac5f2ff",
      "banktransactieIds": [],
      "bedragDifference": null,
      "context": {
        "is": false,
        "something": "else",
      },
      "id": "bef7c096-40b6-4158-b0a5-5b6b813080be",
      "isActive": true,
      "timeUpdated": "2022-05-16T13:00:00.000Z",
      "type": "default",
    },
    {
      "actions": [],
      "alarmId": "1a8ac60f-081f-4bbb-baca-6b6d399ce8db",
      "banktransactieIds": [],
      "bedragDifference": null,
      "context": {
        "isAdmin": true,
        "user": {
          "name": "Koen Brouwer",
        },
      },
      "id": "09c1af54-1054-4a6c-aecd-875fab4ce889",
      "isActive": true,
      "timeUpdated": "2022-05-16T13:00:00.000Z",
      "type": "default",
    },
    {
      "actions": [],
      "alarmId": "738247bc-e6b8-41ee-a09e-a6778822598d",
      "banktransactieIds": [],
      "bedragDifference": null,
      "context": {
        "is": true,
        "something": "else",
      },
      "id": "cd605096-7336-4eb6-b1a7-bc6e1d57e492",
      "isActive": true,
      "timeUpdated": "2022-05-16T13:00:00.000Z",
      "type": "default",
    },
  ],
  "ok": true,
}
`;

exports[`Signal CRUD (operations) Given the Signal exists should return the Signal 1`] = `
{
  "data": {
    "actions": [],
    "alarmId": "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
    "banktransactieIds": [],
    "bedragDifference": "100",
    "context": {
      "is": true,
      "something": "else",
    },
    "id": "02c49b0b-def0-46b2-ae03-9b90430adb94",
    "isActive": true,
    "timeUpdated": "2022-05-16T13:00:00.000Z",
    "type": "default",
  },
  "ok": true,
}
`;

exports[`Signal CRUD (operations) Given the Signal exists should update the Signal 1`] = `
{
  "data": {
    "actions": [],
    "alarmId": "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
    "alarmUuid": "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
    "banktransactieIds": [],
    "banktransactieUuids": [],
    "bedragDifference": "300",
    "context": {
      "is": true,
      "something": "else",
    },
    "id": "02c49b0b-def0-46b2-ae03-9b90430adb94",
    "isActive": true,
    "timeUpdated": "2022-05-16T13:00:00.000Z",
    "type": "default",
  },
  "ok": true,
}
`;
